mlflow/lightgbm.py

```diff
@@ -482,12 +487,9 @@ def log_feature_importance_plot(features, importance, importance_type):
             # training model
             model = original(*args, **kwargs)
 
-            # If early_stopping_rounds is present, logging metrics at the best iteration
+            # If early stopping is activated, logging metrics at the best iteration
             # as extra metrics with the max step + 1.
-            early_stopping_index = all_arg_names.index("early_stopping_rounds")
-            early_stopping = (
-                num_pos_args >= early_stopping_index + 1 or "early_stopping_rounds" in kwargs
-            )
+            early_stopping = model.best_iteration > 0
```

Comment:
`model.best_iteration > 0` can be used to determine whether early stopping is activated.

`model.best_iteration` is initiated as 0 and is assigned with a positive int when `EarlyStoppingException` is raised (this means early stopping is activated, including using the `early_stopping_rounds` param, `lightgbm.early_stopping` callback function, or even a user-defined early_stopping callback that raises `EarlyStoppingException`).
`model.best_iteration` is returned as 0 when early stopping is not activated.

Reference: https://github.com/microsoft/LightGBM/blob/ce486e5b45a6f5e67743e14765ed139ff8d532e5/python-package/lightgbm/engine.py#L226-L263